### DISCUSSION TITLE
Which Collaborative Development Model(s) for this project?

### DISCUSSION THREAD

**Discussion by grant-allan-ctct** [hardbyte/python-can#1696] [General]
Hi, I haven't contributed to a github project before, and I am wondering about how to do it, and specifically how to do it for this project in particular.  

In this github page about [Collaborative Development Models](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/about-collaborative-development-models), two different approaches are mentioned: "fork and pull" and "shared repository".  I can see evidence of "fork and pull" approach in the current pull requests, but I just thought to ask whether we are using just that approach in this project, or maybe both could be OK?  

I have a small change that I would like to make (for [issue 1694](https://github.com/hardbyte/python-can/issues/1694)) and I am wondering whether it's worth a whole new fork just to allow 3 characters to be tweaked in a documentation file? 😎   However, if that's the way forward, then I'll give it a shot.

**Comment by zariiii9003** [comment]
Yes, fork and pull is correct:
1. Create a fork
2. Create a new branch from the `develop` branch
3. Implement your change in that new branch
4. Create a Pull request

**Reply by grant-allan-ctct** [reply]
Sorry, one more question: based on the conversation in [PR 1688](https://github.com/hardbyte/python-can/pull/1688#issuecomment-1815391195), should Step 2 above be to create a new branch from `main`, rather than from `develop`, or have I misunderstood in one of these places?

**Reply by zariiii9003** [reply]
My bad, we renamed `develop` to `main` a few days ago.

**Reply by grant-allan-ctct** [reply]
I am not sure whether [documentation for contributors](https://python-can.readthedocs.io/en/stable/development.html#contributing) requires an update for the move from `develop` to `main`, versus it's already addressed somewhere and not fully flushed through the system yet.

**Reply by zariiii9003** [reply]
Yes, that should be updated 😄

**Reply by grant-allan-ctct** [reply]
I just noticed that https://python-can.readthedocs.io/en/stable/development.html#contributing is still suggesting to arriving developers that they should open any feature requests against the `develop` branch.
